File: schemaorggo/blog.go
package schemaorggo

import "encoding/json"

// Blog see : https://schema.org/Blog
type Blog struct {
	CreativeWork

	typeContext

	// BlogPost see : https://schema.org/blogPost
	// A posting that is part of this blog. Supersedes blogPosts (see: https://schema.org/blogPosts).
	// types : BlogPosting
	BlogPost []*BlogPosting `json:"blogPost,omitempty"`

	// Issn see : https://schema.org/issn
	// The International Standard Serial Number (ISSN) that identifies this serial publication. You can repeat this property to identify different formats of, or the linking ISSN (ISSN-L) for, this serial publication.
	// types : Text
	Issn []string `json:"issn,omitempty"`
}

func (v Blog) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.CreativeWork.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.BlogPost
		if len(v.BlogPost) == 1 {
			value = v.BlogPost[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["blogPost"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.Issn
		if len(v.Issn) == 1 {
			value = v.Issn[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["issn"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v Blog) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "Blog"

	return data, nil
}

func (v Blog) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/blog.php
<?php

class Blog extends CreativeWork implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Blog';
	
	/**
	 * A posting that is part of this blog. Supersedes blogPosts (see: https://schema.org/blogPosts).
	 * see : https://schema.org/blogPost
	 * @var \BlogPosting|\BlogPosting[]
	 */
	public var $blog_post;
	
	/**
	 * The International Standard Serial Number (ISSN) that identifies this serial publication. You can repeat this property to identify different formats of, or the linking ISSN (ISSN-L) for, this serial publication.
	 * see : https://schema.org/issn
	 * @var string|string[]
	 */
	public var $issn;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Blog'
		);
		
		$serialized = so_json_serialize( $this->blog_post );
		if ( ! empty( $serialized ) ) {
			$out['blogPost'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->issn );
		if ( ! empty( $serialized ) ) {
			$out['issn'] = $serialized;
		}
		
		return $out;
	}
}
